feat: Add Dependabot auto-merge workflow

Marenz opened the pull request:
This PR adds a GitHub workflow to automatically manage Dependabot pull requests.

This workflow uses the `ad/dependabot-auto-approve@v1` GitHub Action, which may need to be explicitly whitelisted in the organization's settings.

Marenz committed:
feat: add dependabot auto-merge workflow

Signed-off-by: Mathias L. Baumann <mathias.baumann@frequenz.com>

llucax reviewed (changes requested):
Needs changes to work with protobuf/grpc updates.

Marenz committed:
Update dependabot workflow to ignore grpc/protobuf PRs

Use latest commit (d60f853) with ignore-regex-pr-title feature to skip
grpc and protobuf updates that require manual intervention.

Signed-off-by: Mathias L. Baumann <mathias.baumann@frequenz.com>

Marenz committed:
docs: Fix documentation build warnings

- Disable missing type warnings in docstrings (docstring_options.warn_missing_types: false)
  to suppress warnings from generated gRPC code without type annotations
- Remove unused google/protobuf/empty.proto import from dispatch.proto

This fixes the documentation generation CI failure caused by griffe
detecting missing type annotations in generated gRPC Python code.

Signed-off-by: Mathias L. Baumann <mathias.baumann@frequenz.com>

llucax commented:
> WARNING -  griffe: /opt/hostedtoolcache/Python/3.11.14/x64/lib/python3.11/site-packages/frequenz/api/dispatch/v1/dispatch_pb2_grpc.py:77: No type or annotation for parameter 'channel'

It looks like for API projects we need to add this to `mkdocs.yaml`:

```yaml
plugins:
  # ...
  - mkdocstrings:
      default_handler: python
      handlers:
        python:
          options:
            # ...
            docstring_options:
              warn_missing_types: false
```

Marenz committed:
build: Bump frequenz-repo-config to 0.14.0

Signed-off-by: Mathias L. Baumann <mathias.baumann@frequenz.com>
